archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterAccessesTest.java

```diff
@@ -898,6 +900,19 @@ void call(Target target) {
                 .isEqualTo(call.getTarget().getRawParameterTypes());
     }
 
+    @Test
+    public void identifies_call_origin_if_signature_and_descriptor_deviate() {
+        // Kotlin inline functions cause the creation of extra classes where the signature of the respective method shows
+        // the real types while the descriptor shows the erased types. The erasure of the real types from the signature
+        // does then not match the descriptor in some cases.
+
+        JavaClass javaClass = getOnlyElement(new ClassFileImporter().importLocations(singleton(Location.of(
+                relativeResourceUri(getClass(), "testexamples/MismatchBetweenDescriptorAndSignature.class")))));
```

Comment:
I wonder whether we should somehow conserve the source code from https://github.com/pkubowicz/archunit-accessrecord-bug/blob/main/app/src/main/kotlin/com/example/CountryValidator.kt – and if it's only in the commit message.

Reply:
Good idea, I added it as a comment. Doing this also made me strip down the example more :joy: